BeagleBone_black/src/sensors/vl6180.cpp

```diff
@@ -123,12 +137,27 @@ void VL6180::turnOn()
   // Perform a single recalibration
   writeByte(kSysrangeVhvRecalibrate, 0x01);
 
+  writeByte(kSystemInterruptConfigGpio, 0x24);
+
   // Set max convergence time (Recommended default 50ms)
   uint8_t time_ms = 50;  // changes here
   setMaxConvergenceTime(time_ms);
 
+  // Clear interrupt
+  writeByte(kSystemInterruptClear, 0x01);
+
+  // writeByte(kSystemFreshOutOfReset, 0x00);
+
   if (isOnline()) {
     log_.INFO("VL6180", "Sensor is online");
+    Proximity proxi;
+    getData(&proxi);
+    getData(&proxi);
+    Thread::sleep(100);
+    if (timeout_) setContinuousRangingMode();
+    Thread::sleep(100);
+    getData(&proxi);
+    if (timeout_) setContinuousRangingMode();
```

Comment:
Shouldn't we call `getData` again after this? If the `if` executes, we will have `timeout_ == false` and `is_online_ == false` when `turnOn()` returns even if the sensor is online and fully functional. Is that okay?